Tests/14.12.2021/finiteStateMachine.c

```diff
@@ -0,0 +1,119 @@
+#include <stdbool.h>
+#include <string.h>
+#include "finiteStateMachine.h"
+
+char getNextChar(const char* string, int* index)
+{
+    if (*index < 0 || *index > strlen(string))
+    {
+        return '\0';
+    }
+    char buffer = string[*index];
+    ++* index;
```

Comment:
```suggestion
    ++*index;
```
Унарные операторы пробелами не отделяются